packages/gatsby-theme-docs/src/hooks/use-active-section.js

```diff
@@ -15,16 +15,32 @@ const getSectionElements = () =>
 
 const offset = calculateOffset();
 
+const isInViewport = (element) => {
+  const rect = element.getBoundingClientRect();
+
+  return (
+    (rect.top >= 0 &&
+      rect.top <
+        (window.innerHeight || document.documentElement.clientHeight)) ||
+    (rect.top < 0 && rect.bottom >= 0)
+  );
+};
+
 const useActiveSelection = () => {
   const [activeSection, setActiveSection] = React.useState();
   const onScroll = React.useCallback(() => {
     const sectionElements = getSectionElements();
     let nextActiveSection;
+    const visibleSections = [];
     sectionElements.forEach((section) => {
```

Comment:
Just to document my confusion further:  The forEach over the sections looks like the last section in the `sectionElements` result is the one that ends up being highlighted through `setActiveSection`, but our observation is that the topmost one is highlighted if multiple are in view.